usecode/intrinsics.cc

```diff
@@ -3058,6 +3059,35 @@ USECODE_INTRINSIC(play_sound_effect2) {
 	return no_ret;
 }
 
+USECODE_INTRINSIC(play_scene) {
+	if (num_parms < 1) {
+		return no_ret;
+	}
+
+	const char* scene_name_str = parms[0].get_str_value();
+	if (!scene_name_str) {
+		return no_ret;
+	}
+
+	std::string scene_name(scene_name_str);
+	if (scene_available(scene_name)) {
+		// fade outs in scenes will screw up the palette, so we save current
+		// palette index and restore it after the scene is played.
+		int prev_palette = gwin->get_pal()->get_palette_index();
+		// Pause the queue.
+		gwin->get_tqueue()->pause(Game::get_ticks());
+		play_scene(scene_name);
+
+		// Restore the palette index.
+		gwin->get_pal()->load(PALETTES_FLX, PATCH_PALETTES, prev_palette);
+		gwin->get_pal()->apply(true);
```

Comment:
Maybe fading to black before the scene, following by fading the cut-scene to black, then fading from black to old palette? This could even be another parameter to opt in/out of this fading.

Reply:
Thanks, I'll do the txt file as you wrote and will try with the palette as you suggested

Reply:
As the text scenes change the palette, I need to keep track of it, but I moved this part off to playscene.cc. Fade_out/fade_in added anyway as optional bools and autogenerated exult_intrinsics.txt. By chance I stumbled over the weird text in teleport_to_saved_pos and corrected that, too. Saves me another commit later on, but I can keep it clean. The seventowers web file changes are lined up as well.